com.microsoft.mrtk.uxcore/Keyboard/NonNative/NonNativeValueKey.cs

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using TMPro;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.UX
+{
+    /// <summary>
+    /// Class representing a value key in the non native keyboard
+    /// </summary>
+    public class NonNativeValueKey : NonNativeKey
+    {
+        private string currentValue;
+
+        /// <summary>
+        /// The current string value of this value key. Note the value may change based on the shift status of the keyboard.
+        /// </summary>
+        public string CurrentValue
```

Comment:
Is there way to use that crazy `[field:` syntax you used in the abstract class for this?

Reply:
Nvm, you can't do this with a custom setter.